src/overlays/actors/ovl_Bg_Spot05_Soko/z_bg_spot05_soko.c

```diff
@@ -21,17 +32,77 @@ const ActorInit Bg_Spot05_Soko_InitVars = {
     (ActorFunc)BgSpot05Soko_Update,
     (ActorFunc)BgSpot05Soko_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot05_Soko/BgSpot05Soko_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot05_Soko/BgSpot05Soko_Destroy.s")
+static InitChainEntry initChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+static Gfx* dListTbl[] = {
+    0x06000840,
+    0x06001190,
+};
+
+static void BgSpot05Soko_Init(BgSpot05Soko* this, GlobalContext* globalCtx) {
+
+    Actor* thisx = &this->dyna.actor;
+    u32 sp24[2];
+    sp24[0] = 0;
+    Actor_ProcessInitChain(thisx, initChain);
+    this->switchFlag = (thisx->params >> 8) & 0xFF;
+    thisx->params &= 0xFF;
+    DynaPolyInfo_SetActorMove(thisx, DPM_UNK);
+    if (thisx->params == 0) {
+        DynaPolyInfo_Alloc(&D_06000918, &sp24);
+        if (LINK_IS_ADULT) {
+            Actor_Kill(thisx);
+        } else {
+            this->actionFunc = &func_808AE5A8;
+        }
+    } else {
+        DynaPolyInfo_Alloc(&D_060012C0, &sp24);
+        if (Flags_GetSwitch(globalCtx, this->switchFlag) != 0) {
+            Actor_Kill(thisx);
+        } else {
+            this->actionFunc = &func_808AE5B4;
+            thisx->flags |= 0x10;
+        }
+    }
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, thisx, sp24[0]);
+}
+
+static void BgSpot05Soko_Destroy(BgSpot05Soko* this, GlobalContext* globalCtx) {
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+}
+
+static void func_808AE5A8(BgSpot05Soko* this, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot05_Soko/func_808AE5A8.s")
+static void func_808AE5B4(BgSpot05Soko* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->dyna.actor;
+    if (Flags_GetSwitch(globalCtx, this->switchFlag)) {
+        Audio_PlaySoundAtPosition(globalCtx, &thisx->posRot.pos, 0x1E, NA_SE_EV_METALDOOR_CLOSE);
+        Actor_SetHeight(thisx, 50.0f);
+        func_80080480(globalCtx, thisx);
+        this->actionFunc = &func_808AE630;
+        thisx->speedXZ = 0.5f;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot05_Soko/func_808AE5B4.s")
+static void func_808AE630(BgSpot05Soko* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->dyna.actor;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot05_Soko/func_808AE630.s")
+    thisx->speedXZ *= 1.5f;
+    if (Math_ApproxF(&thisx->posRot.pos.y, thisx->initPosRot.pos.y - 120.0f, thisx->speedXZ) != 0) {
+        Actor_Kill(thisx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot05_Soko/BgSpot05Soko_Update.s")
+static void BgSpot05Soko_Update(BgSpot05Soko* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->dyna.actor;
+    this->actionFunc(thisx, globalCtx);
```

Comment:
Action functions should normally take the specific actor struct (BgSpot05Soko in this case) not the common Actor struct, so I don't think thisx is needed at all here.